server/api/user/orders.js

```diff
@@ -29,4 +29,31 @@ router.get('/:id', async (req, res, next) => {
   }
 })
 
+router.post('/', async (req, res, next) => {
+  try {
+    const order = await Order.create({
+      status: req.body.status,
+      subtotal: req.body.subtotal,
+      shippingAddress: req.body.shippingAddress,
+      email: req.body.email,
+    });
+    const user = await User.findById(req.session.passport.user);
+    order.setUser(user);
+    req.body.productsInCart.map( async (productInCart)  => {
```

Comment:
instead of awaiting each product, you could wrap this in a Promise.all() for it to occur simultaneously